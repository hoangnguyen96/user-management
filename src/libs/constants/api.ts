export const API_NAME = "user_management";

export enum ENDPOINTS {
  USER = "user",
  PRODUCT = "product",
}

export enum METHOD {
  GET = "GET",
  POST = "POST",
  PUT = "PUT",
  PATCH = "PATCH",
  DELETE = "DELETE",
}

export const OPTIONS = ["Edit", "Delete"];
