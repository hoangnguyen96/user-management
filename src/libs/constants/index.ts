export * from "./api";
export * from "./dashboard";
export * from "./error-message";
export * from "./mutation-key";
export * from "./query-key";
export * from "./query-key";
export * from "./routes";
export * from "./selection";
export * from "./user";
export * from "./validate";
export * from "./key-token";
