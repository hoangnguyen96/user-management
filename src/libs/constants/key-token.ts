export const TOKEN_KEY = "user-manage";
