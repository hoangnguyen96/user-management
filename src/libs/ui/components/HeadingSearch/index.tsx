import { ChangeEvent, Dispatch, SetStateAction, memo } from "react";
import { Box, Stack, Typography } from "@mui/material";
import { Button, SearchBar } from "../Common";
import { Selection } from "../Selection";

interface HeadingSearchProps {
  textSearch: string;
  title: string;
  subTitle: string;
  list: {
    value: string;
    name: string;
  }[];
  isProduct?: boolean;
  backgroundColor?: string;
  handleOnChangeText: (e: ChangeEvent<HTMLInputElement>) => void;
  onOpenModalProduct?: () => void;
  setSelectionValue: Dispatch<SetStateAction<string>>;
}

const HeadingSearch = ({
  title,
  subTitle,
  textSearch,
  list,
  backgroundColor,
  isProduct = false,
  handleOnChangeText,
  onOpenModalProduct,
  setSelectionValue,
}: HeadingSearchProps) => (
  <Stack
    flexDirection="row"
    alignItems="center"
    justifyContent="space-between"
    p="30px 40px 40px"
  >
    <Box>
      <Typography variant="h4">{title}</Typography>
      <Typography variant="subtitle2" color="#16c098">
        {subTitle}
      </Typography>
    </Box>
    <Stack
      flexDirection="row"
      alignItems="center"
      justifyContent="space-between"
      gap="16px"
    >
      <SearchBar
        value={textSearch}
        backgroundSearch={backgroundColor}
        onChange={handleOnChangeText}
      />
      <Selection
        list={list}
        backgroundColor={backgroundColor}
        setSelectionValue={setSelectionValue}
      />
      {isProduct && (
        <Button
          data-testid="modal-add-product"
          variant="contained"
          label="Add"
          onClick={onOpenModalProduct}
        />
      )}
    </Stack>
  </Stack>
);

export default memo(HeadingSearch);
