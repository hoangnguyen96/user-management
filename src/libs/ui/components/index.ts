// Common
export * from "./Common";

// Components
export { default as HeadingContent } from "./HeadingContent";
export { default as LazyFallbackContent } from "./LazyFallbackContent";
export { default as Logo } from "./Logo";
export { default as Footer } from "./Footer";
export { default as FormController } from "./FormController";
export { default as ListCustomersContent } from "./ListCustomersContent";
export { default as ListProductContent } from "./ListProductContent";
export { default as NavigationBar } from "./NavigationBar";
export { default as Temporary } from "./Temporary";
export { default as TitleContent } from "./TitleContent";
export { default as MenuBar } from "./MenuBar";
export { default as ErrorBoundary } from "./ErrorBoundary";
export { default as ModalCreateUserSuccess } from "./Modal/ModalCreateUserSuccess";
