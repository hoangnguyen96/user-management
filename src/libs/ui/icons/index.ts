export * from "./HelpIcon";
export * from "./IncomeIcon";
export * from "./KeyIcon";
export * from "./Members";
export * from "./ProductIcon";
export * from "./PromoteIcon";
export * from "./Screen";
export * from "./SearchIcon";
export * from "./TotalCustomers";
export * from "./UserIcon";
