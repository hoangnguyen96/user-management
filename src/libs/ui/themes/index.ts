import { createTheme } from "@mui/material/styles";
import { breakpointsBase } from "./breakpoints";
import { colors } from "./colors";
import { typographyBase } from "./typography";

export const themeDefault = createTheme({
  breakpoints: breakpointsBase,
  palette: colors,
  typography: typographyBase,

  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          textTransform: "capitalize",
        },
      },
      variants: [
        {
          props: { variant: "contained" },
          style: {
            color: "White",
          },
        },
        {
          props: { variant: "activate" },
          style: {
            backgroundColor: "rgba(22, 192, 152, 0.38)",
            color: "#008767",
            border: "1px solid #00b087",
            "&:hover": {
              backgroundColor: "rgba(22, 192, 152, 0.62)",
            },
          },
        },
        {
          props: { variant: "inactivate" },
          style: {
            backgroundColor: "#ffc5c5",
            color: "#df0404",
            border: "1px solid #df0404",
            "&:hover": {
              backgroundColor: "rgb(253 170 170)",
            },
          },
        },
        {
          props: { variant: "normal" },
          style: {
            backgroundColor: " rgba(245, 245, 245, 1)",
            color: "rgba(64, 75, 82, 1)",
            border: "1px solid #eeeeee",
            "&:hover": {
              backgroundColor: "rgb(219 219 219)",
            },
          },
        },
      ],
      defaultProps: {
        size: "medium",
      },
    },

    MuiTextField: {
      styleOverrides: {
        root: ({ theme }) => ({
          "& .MuiOutlinedInput-root": {
            borderRadius: "12px",
            fontSize: "14px",
            backgroundColor: theme.palette.common.white,
            "& .css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input": {
              padding: "12px",
            },
            "& fieldset": {
              borderColor: "#b5b7C0",
            },
            "&:hover fieldset": {
              borderColor: theme.palette.primary.main,
            },
            "&.Mui-focused fieldset": {
              borderColor: theme.palette.primary.main,
            },
          },
        }),
      },
    },

    MuiSelect: {
      styleOverrides: {
        root: {
          width: "auto",
          flex: 1,
        },
      },
    },

    MuiFormControl: {
      styleOverrides: {
        root: {
          flexDirection: "row",
          justifyContent: "center",
          alignItems: "center",
          backgroundColor: "#f9fbff",
          borderRadius: 12,
          minWidth: 154,
          "& .css-wama2x-MuiFormLabel-root-MuiInputLabel-root": {
            position: "unset",
            transform: "unset",
            left: "unset",
            top: "unset",
          },
          "& .css-cb5d54-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root":
            {
              "& fieldset": {
                border: 0,
              },
              borderColor: "transparent",
              borderWidth: 0,
            },
          "& .css-wxw8j3-MuiFormLabel-root-MuiInputLabel-root": {
            position: "unset",
            transform: "unset",
            left: "unset",
            top: "unset",
            flex: 0.8,
            padding: 10,
            paddingRight: 5,
            fontSize: "12px",
            textAlign: "right",
          },
          "& .css-zbhh1g-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input.MuiSelect-select":
            {
              padding: "11px 32px 11px 0",
              fontSize: "12px",
              fontWeight: 500,
            },
          "& .css-cb5d54-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root.Mui-focused .MuiOutlinedInput-notchedOutline":
            {
              borderColor: "transparent",
              borderWidth: 0,
            },
          "& .css-10d36jd-MuiSvgIcon-root-MuiSelect-icon, & .css-5ajf1d-MuiSvgIcon-root-MuiSelect-icon":
            {
              width: 20,
              height: 20,
            },
          "& .css-1j5jes1-MuiFormLabel-root.Mui-focused": {
            color: "rgba(0, 0, 0, 0.6)",
          },
        },
      },
    },

    MuiPaper: {
      styleOverrides: {
        root: {
          "& .css-ddqxy4-MuiButtonBase-root-MuiMenuItem-root": {
            fontSize: "12px",
          },
        },
      },
    },

    MuiListItemButton: {
      styleOverrides: {
        root: ({ theme }) => ({
          borderRadius: theme.spacing(1),
          "&.Mui-selected": {
            backgroundColor: theme.palette.primary.main,
            color: theme.palette.common.white,
          },
          "&.Mui-selected:hover": {
            backgroundColor: theme.palette.primary.main,
            color: theme.palette.common.white,
          },
        }),
      },
    },

    MuiLink: {
      styleOverrides: {
        root: ({ theme }) => ({
          backgroundColor: "transparent",
          fontSize: "14px",
          textDecoration: "none",
          color: theme.palette.primary.contrastText,
          "&:hover": {
            cursor: "pointer",
          },
        }),
      },
    },

    MuiPagination: {
      styleOverrides: {
        root: ({ theme }) => ({
          "& .MuiPagination-ul": {
            "& .MuiButtonBase-root.MuiPaginationItem-root": {
              fontSize: "12px",
              fontWeight: 500,
              backgroundColor: "#f5f5f5",
              minWidth: "25px",
              height: "24px",
              margin: "0 5px",
              "&.Mui-selected": {
                backgroundColor: theme.palette.primary.main,
                color: "white",
              },
              "& .MuiSvgIcon-root.MuiPaginationItem-icon": {
                fontSize: "12px",
              },
            },
          },
        }),
      },
    },

    MuiContainer: {
      styleOverrides: {
        root: {
          maxWidth: "100%",
          paddingLeft: 0,
          paddingRight: 0,
          "@media (min-width:600px)": {
            paddingLeft: 0,
            paddingRight: 0,
          },
          "@media (min-width:960px)": {
            paddingLeft: 0,
            paddingRight: 0,
          },
          "@media (min-width:1280px)": {
            paddingLeft: 0,
            paddingRight: 0,
          },
          "@media (min-width:1440px)": {
            maxWidth: "1440px",
          },
        },
      },
    },
  },
});
