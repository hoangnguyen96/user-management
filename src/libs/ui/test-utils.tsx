import { PropsWithChildren, ReactElement, ReactNode } from "react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { BrowserRouter } from "react-router-dom";
import { RenderOptions, render } from "@testing-library/react";

// ThemeProvider
import { ThemeProvider } from "@mui/material";

// Config
import { themeDefault } from "./themes";

const AllTheProviders = ({ children }: PropsWithChildren) => {
  return <ThemeProvider theme={themeDefault()}>{children}</ThemeProvider>;
};

export const wrapper = ({ children }: { children: ReactNode }) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  return (
    <ThemeProvider theme={themeDefault()}>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </ThemeProvider>
  );
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const customRenderWithQueryClient = (
  ui: ReactElement,
  options?: RenderOptions
) => render(ui, { wrapper, ...options });

export const WrapperRouter = ({ children }: { children: ReactNode }) => (
  <BrowserRouter>{children}</BrowserRouter>
);

export { customRender as render };
export { customRenderWithQueryClient as renderWithQueryClient };
