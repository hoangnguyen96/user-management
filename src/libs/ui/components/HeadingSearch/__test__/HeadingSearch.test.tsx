import { render } from "@app/ui/test-utils";
import { fireEvent, waitFor } from "@testing-library/react";

// Constants
import { SORT_DATA_CUSTOMERS } from "../../../../constants";

// Component
import HeadingSearch from "..";

describe("HeadingSearch component", () => {
  const props = {
    title: "All Customers",
    subTitle: "Active Members",
    textSearch: "search term",
    list: SORT_DATA_CUSTOMERS,
    setSelectionValue: jest.fn(),
    handleOnChangeText: jest.fn(),
    onOpenModalProduct: jest.fn(),
  };

  it("Should render snapshot correctly", () => {
    expect(render(<HeadingSearch {...props} />)).toMatchSnapshot();
  });

  it("Should renders SearchBar component with correct value and onChange handler", () => {
    const { getByRole } = render(<HeadingSearch {...props} />);
    const searchBar = getByRole("textbox") as HTMLInputElement;
    expect(searchBar).toBeInTheDocument();
    expect(searchBar.value).toBe("search term");

    fireEvent.change(searchBar, { target: { value: "new search term" } });
    expect(props.handleOnChangeText).toHaveBeenCalledTimes(1);
  });

  it("Should renders Add button and calls onOpenModalProduct on click", () => {
    const { getByTestId } = render(
      <HeadingSearch {...props} isProduct={true} />
    );
    const addButton = getByTestId("modal-add-product");
    expect(addButton).toBeInTheDocument();

    fireEvent.click(addButton);
    expect(props.onOpenModalProduct).toHaveBeenCalledTimes(1);
  });

  it("Should renders Selection component with correct list and onChange handler", async () => {
    const { getByTestId, getByRole, getByLabelText, getByText } = render(
      <HeadingSearch {...props} />
    );
    const selection = getByTestId("change-selection");
    expect(selection).toBeInTheDocument();

    fireEvent.mouseDown(getByLabelText(/short by:/i));

    await waitFor(() => getByRole("listbox"));

    const newOption = getByText("Phone");
    fireEvent.click(newOption);

    expect(props.setSelectionValue).toHaveBeenCalledWith("phoneNumber");
  });
});
