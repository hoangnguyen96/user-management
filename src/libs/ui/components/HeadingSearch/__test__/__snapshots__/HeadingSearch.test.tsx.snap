// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingSearch component Should render snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiStack-root css-133lg72-MuiStack-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-15gon46-MuiTypography-root"
          >
            All Customers
          </h4>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-1ge8cc0-MuiTypography-root"
          >
            Active Members
          </h6>
        </div>
        <div
          class="MuiStack-root css-qoybat-MuiStack-root"
        >
          <div
            class="MuiBox-root css-79elbk"
          >
            <div
              class="MuiStack-root css-1ibunwk-MuiStack-root"
            >
              <svg
                fill="none"
                height="24px"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 19C15.4183 19 19 15.4183 19 11C19 6.58172 15.4183 3 11 3C6.58172 3 3 6.58172 3 11C3 15.4183 6.58172 19 11 19Z"
                  stroke="#7e7e7e"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
                <path
                  d="M21 21L16.65 16.65"
                  stroke="#7e7e7e"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-94omkc-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-label="search"
                class="MuiInputBase-input MuiInput-input css-15e8ec1-MuiInputBase-input-MuiInput-input"
                placeholder="Search…"
                type="text"
                value="a"
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root css-2iaed4-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-wxw8j3-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="demo-simple-select-label"
            >
              Short by:
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-cb5d54-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              data-testid="change-selection"
            >
              <div
                aria-controls=":r0:"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-zbhh1g-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="demo-simple-select"
                role="combobox"
                tabindex="0"
              >
                -select-
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="default"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-10d36jd-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="KeyboardArrowDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiStack-root css-133lg72-MuiStack-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-15gon46-MuiTypography-root"
        >
          All Customers
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-1ge8cc0-MuiTypography-root"
        >
          Active Members
        </h6>
      </div>
      <div
        class="MuiStack-root css-qoybat-MuiStack-root"
      >
        <div
          class="MuiBox-root css-79elbk"
        >
          <div
            class="MuiStack-root css-1ibunwk-MuiStack-root"
          >
            <svg
              fill="none"
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 19C15.4183 19 19 15.4183 19 11C19 6.58172 15.4183 3 11 3C6.58172 3 3 6.58172 3 11C3 15.4183 6.58172 19 11 19Z"
                stroke="#7e7e7e"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M21 21L16.65 16.65"
                stroke="#7e7e7e"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-94omkc-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input MuiInput-input css-15e8ec1-MuiInputBase-input-MuiInput-input"
              placeholder="Search…"
              type="text"
              value="a"
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root css-2iaed4-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-wxw8j3-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            id="demo-simple-select-label"
          >
            Short by:
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-cb5d54-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            data-testid="change-selection"
          >
            <div
              aria-controls=":r0:"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-label demo-simple-select"
              class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-zbhh1g-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="demo-simple-select"
              role="combobox"
              tabindex="0"
            >
              -select-
            </div>
            <input
              aria-hidden="true"
              aria-invalid="false"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="default"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-10d36jd-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="KeyboardArrowDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
